### ISSUE TITLE
Failed to build due to rust error[E0557]

### ISSUE THREAD

**Issue by rsahita** [diodesign/diosix#20]
https://github.com/diodesign/supervisor-riscv.git had a dependency on linked_list_allocator version = "0.8.11" which had a dependency on lock_api 0.3.0 which was causing the build error below.

The failed build due to an "error[E0557]: feature has been removed" for #![feature(const_fn)]

updating the dependency for linked_list_allocator to version 0.9.1 moves the dependent lock_api lib up to 0.4.5 like so:
├── linked_list_allocator v0.9.1
│   └── spinning_top v0.2.4
│       └── lock_api v0.4.5
│           └── scopeguard v1.1.0

This fixes the build.
suggest also updating the riscv dependency to 0.7.0 
As well as spin to 0.9.2 (which also has a lock_api dependency)

thanks!

**Comment by diodesign** [comment]
Thanks a lot for the feedback! I'm just working on a side project related to Diosix and RISC-V, which does involve changes to Diosix's build system, and when that's working, I'll turn to the dependencies.

**Comment by rsahita** [comment]
Sure. Do you plan on adding RV Hypervisor extension support to diosix?

Ravi

**Comment by diodesign** [comment]
Yes: I want it to be able to take advantage of the H extension if present, and PMP if not -- thanks for the interest, it's a big motivator for me

**Comment by FrancoAuri1996** [comment]
Hi, how can i update the dependencies?

**Comment by diodesign** [comment]
Hey, we've move to Zig now, so I can't answer questions about Rust. We maintain a rust branch for legacy reasons but future development is in Zig.
